tests/validation/tests/v3_api/test_hosted_eks_cluster.py

```diff
@@ -99,27 +102,74 @@ def test_create_hosted_eks_cluster_2():
     create_and_validate_eks_cluster(cluster_config)
 
 
+@ekscredential
+def test_eks_v2_create_import_cluster():
+    ec2_cloud_credential = get_aws_cloud_credential()
+    display_name = create_resources_eks()
+    cluster_name = random_test_name("test-auto-eks")
+
+    eks_config_temp = {
+        "amazonCredentialSecret": ec2_cloud_credential.id,
+        "displayName": display_name,
+        "imported": True,
+        "privateAccess": False,
+        "publicAccess": False,
+        "region": EKS_REGION,
+        "secretsEncryption": False,
+        "type": "eksclusterconfigspec"
+    }
+
+    cluster_config = {
+        "eksConfig": eks_config_temp,
+        "name": cluster_name,
+        "type": "cluster",
+        "dockerRootDir": "/var/lib/docker",
+        "enableNetworkPolicy": False,
+        "enableClusterAlerting": False,
+        "enableClusterMonitoring": False
+    }
+    create_and_validate_eks_cluster(cluster_config,
+                                    imported=True)
+
+
+def create_resources_eks():
+    cluster_name = resource_prefix + "-ekscluster"
+    AmazonWebServices().create_eks_cluster(cluster_name)
+    AmazonWebServices().wait_for_eks_cluster_state(cluster_name, "ACTIVE")
+    IMPORTED_EKS_CLUSTERS.append(cluster_name)
```

Comment:
Can we add the cluster name to the array before waiting for state? If the state doesn't get to the expected state before timeout, the cluster will not be deleted.

Reply:
Addressed in changes. Cluster name is added before waiting for state.